crypto/asn1/a_digest.c

```diff
@@ -76,7 +76,7 @@ int asn1_item_digest_with_libctx(const ASN1_ITEM *it, const EVP_MD *md,
 #endif
             fetched_md = EVP_MD_fetch(libctx, EVP_MD_name(md), propq);
     }
-     if (fetched_md == NULL)
+     if (fetched_md == NULL || i < 0)
```

Comment:
Please add the check for `i` ASN1_item_i2d() call instead.

This line has a space too much.

Please document this function.

Reply:
Moved the line up... Documentation is outside the scope.